Env diff changes

rakete opened the pull request.

rakete committed:
add workflow to upload env-diff.sh to oss

rakete committed:
add env-diff.sh with changes

- double quoting vs. double bracket syntax
- use ${var} for all vars
- remove hardcoded paths, output to stdout instead
- also output to stderr we can output to both file and github actions

rakete committed:
fix syntax error in deploy.yml

Charlotte-z commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,21 @@
+        enpoint: oss-eu-central-1.aliyuncs.com
+        config_dir: /oss-config
+
+    - name: Download env-diff.sh script
```
Download env-diff.sh script

suggpest to 

Upload env-diff.sh script

rakete committed:
change "Download" to "Upload", cause thats what we're doing

rakete committed:
upload env-diff.sh as env-diff-new.sh so I can gradually make changes using it

- this way I won't be breaking existing scripts using the old one
until I am ready to
